pkg/api/mlflow/service/artifact/service.go

```diff
@@ -58,3 +59,33 @@ func (s Service) ListArtifacts(
 
 	return run.ArtifactURI, artifacts, nil
 }
+
+// GetArtifact handles business logic of `GET /get-artifact` endpoint.
+func (s Service) GetArtifact(
+	ctx context.Context, req *request.GetArtifactRequest,
+) (io.ReadCloser, error) {
+	if err := ValidateGetArtifactRequest(req); err != nil {
+		return nil, err
+	}
+
+	run, err := s.runRepository.GetByID(ctx, req.GetRunID())
+	if err != nil {
+		return nil, api.NewInternalError("unable to find run '%s': %s", req.GetRunID(), err)
+	}
+	if run == nil {
+		return nil, api.NewResourceDoesNotExistError("unable to find run '%s'", req.GetRunID())
+	}
+	artifactStorage, err := s.artifactStorageFactory.GetStorage(run.ArtifactURI)
+	if err != nil {
+		return nil, api.NewInternalError("run with id '%s' has unsupported artifact storage", run.ID)
+	}
+
+	artifactReader, err := artifactStorage.GetArtifact(
```

Comment:
will be nice to return here just a prepared bytes. also please change `artifactReader` to be `artifactObject`.

Reply:
but what if it's big? I think we should deal in Reader/Writer instead of slice -- see above comment for alternative.

Reply:
I think we can try and see. @jgiannuzzi will be nice to have your opinion.

Reply:
I agree with @suprjinx that it's best to return an `io.ReadCloser` here so we can stream the artifact's bytes back to the client without making intermediate copies

Reply:
ok.